src/os/ui/draw/draw.js

```diff
@@ -1,8 +1,166 @@
+
+
+/**
+ * Build a list of features to represent the search grid that intersects with the drawn geometry
+ *
+ * @param {ol.Feature} feature The source feature to trace.
+ * @param {osx.ui.draw.GridOptions} options Config object for grid generation
+ * @return {Array.<ol.Feature>}
+ *
+ * @suppress {accessControls} To allow direct access to feature metadata.
+ */
+os.ui.draw.getGridFromFeature = function(feature, options) {
+  if (!feature || !options) return null;
+
+  var detail = options.detail;
+  var max = options.max;
+
+  var features = [];
+  var geo = feature.getGeometry();
+
+  geo.toLonLat(); // convert to lon/lat so grid 'detail' (in degrees) can be applied easily
+  var extent = geo.getExtent(); // build a simplified box around the search geometry
+  geo.osTransform(); // undo transformation so copyFeature() succeeds
+
+  // don't continue building the grid if it would create too many boxes
+  if ((Math.ceil(Math.abs(extent[2] - extent[0]) / detail) *
+      Math.ceil(Math.abs(extent[3] - extent[1]) / detail)) > max) return null;
+
+  // snap box coordinates to the nearest "detail" degrees
+  var snap = function(l, d, ceil) {
+    if ((l % d) != 0.0) return Math.floor(l / d) * d + (ceil ? d : 0.0);
+    return l;
+  };
+
+  extent[0] = snap(extent[0], detail, false); // lon min
+  extent[1] = snap(extent[1], detail, false); // lat min
+  extent[2] = snap(extent[2], detail, true); // lon max
+  extent[3] = snap(extent[3], detail, true); // lat max
+
+  var cfg = {
+    lon: {
+      n: (extent[0] < extent[2]) ? extent[0] : extent[2],
+      x: (extent[0] < extent[2]) ? extent[2] : extent[0]
+    },
+    lat: {
+      n: (extent[1] < extent[3]) ? extent[1] : extent[3],
+      x: (extent[1] < extent[3]) ? extent[3] : extent[1]
+    }
+  }; // sort the data so the loops later are faster
+
+  var prop = {};
+  goog.object.extend(prop, feature.values_);
+  delete prop['geometry'];
+  delete prop['title'];
+  delete prop['_node'];
+
+  // TODO research a better, faster way to create/approximate the grid
+  // build detail x detail degree boxes that fit the "snapped" extent
+  for (var lon = cfg.lon.n; lon < cfg.lon.x; lon += detail) {
+    for (var lat = cfg.lat.n; lat < cfg.lat.x; lat += detail) {
+      var gridExtent = ol.proj.transformExtent(
+          [lon, lat, lon + detail, lat + detail],
+          os.proj.EPSG4326,
+          os.map.PROJECTION);
+
+      // only add this to the grid if the original polygon intersects it
+      if (feature.getGeometry().intersectsExtent(gridExtent)) {
+        features.push(os.ui.draw.gridFeatureFromExtent_(gridExtent, prop, options));
+      }
+    }
+  }
+
+  // TODO build a trace of the grid and save that to the Drawing Layers
+
+  return features;
+};
+
+
+/**
+ * Helper function; gets a numeric representation of the JSON setting
+ *
+ * @param {string} key
+ * @param {number} defaultValue
+ * @return {number}
+ */
+os.ui.draw.getGridSetting = function(key, defaultValue) {
```

Comment:
Let's get rid of the settings stuff entirely in here. Anything that wants to use `os.ui.draw.getGridFromFeature()` should be responsible for creating its own options.

Reply:
I left this to simplify my code, but valid alternatives are to:
1. Add a "getAs()" function to settings.js which would accept a casting function, OR
2. Add a similar "toFloat()" function to all the places that need it to meet the value tests

I didn't want to do the first option since it's modifying a very high-profile file; and it's overkill for this use-case.
I didn't want to do the second option since it'd end up creating a lot of duplicated casting code as more plugins use the GridOptions.

Reply:
If I were using this from a plugin, I would do this (since you added an extern for the `GridOptions` type):

```
var options = /** @type {undefined|osx.GridOptions} */ (os.settings.get('whatever'));
if (options) {
  os.ui.draw.getGridFromFeature(feature, options);
}
```

The test using `getGridSetting` is unnecessary.

Reply:
That's what I ended up doing -- should be on the same page now.